src/GitHubTools/Functions/Public/Update-AzurePipelineDependencies.ps1

```diff
@@ -0,0 +1,93 @@
+<#
+
+.SYNOPSIS
+Updates the refs of the azure-pipelines.yml file's repository resources to the latest
+
+.DESCRIPTION
+Updates the refs of the azure-pipelines.yml file's repository resources to the latest
+
+.PARAMETER GitHubOrganisation
+The GitHub organisation
+
+.PARAMETER RepositoryPrefix
+The prefix of repositories to create new branches with updated pipeline dependencies
+
+.EXAMPLE
+Set-GitHubSessionInformation -PatToken <not-a-real-pat-token>
+Update-AzurePipelineDependencies -GitHubOrganisation MyOrganisation -RepositoryPrefix foo-
+
+.NOTES
+powershell-yaml powershell module is required. Install-Module -name powershell-yaml
+The powershell-yaml module is no longer being maintained https://github.com/aaubry/YamlDotNet/discussions/689
+
+#>
+function Update-AzurePipelineDependencies {
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute("PSReviewUnusedParameter", "RepositoryPrefix", Justification = "False positive as rule does not know that Where-Object operates within the same scope")]
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute("PSUseDeclaredVarsMoreThanAssignments", "UpdatePipelineYml", Justification="False positive as rule does not know that variable is updated in ForEach-Object")]
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute("PSUseShouldProcessForStateChangingFunctions", "")]
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute("PSUseSingularNouns", "")]
+    param(
+        [Parameter(Mandatory = $false)]
+        [String]$GitHubOrganisation = "SkillsFundingAgency",
+        [Parameter(Mandatory = $false)]
+        [String]$RepositoryPrefix = "das-"
+    )
+
+    $Repos = Get-GitHubRepo -GitHubOrg $GitHubOrganisation | Where-Object { $_.name -match "^$RepositoryPrefix.+" -and $_.isArchived -eq $False }
+
+    $Repos | ForEach-Object {
+        $Repo = $_
+
+        Write-Host "Scanning $($Repo.name)"
+
+        $PipelineYml = Get-GitHubRepoFileContent -GitHubOrganisation $GitHubOrganisation -RepositoryName $Repo.name -FilePath "azure-pipelines.yml"
+
+        if (!$PipelineYml) { return }
```

Comment:
- write a warning if no `azure-pipelines.yml` retrieved
- use `continue` if changing this to a `foreach` loop